EIPS/eip-3664.md

```diff
@@ -15,403 +15,369 @@ requires: 20, 1155
-```
-
-Attribute upgrade sub level for nft. only called by contract operator.
-
-- Param1: attribute owned id
-- Param2: attribute recipient id
-- Param3: attribute sub level
-
-### IEvolutiveAttribute
-
-```
-pragma solidity ^0.8.0;
-
-interface IEvolutiveAttribute {
-    /**
-     * @dev Returns the name of the attribute .
-     */
-    function name(uint256 _attrId) external view returns (string memory);
-
+/**
+ * @dev Interface for the updatable functions from the ERC3664.
+ */
+interface IERC3664Updatable is IERC3664Metadata {
     /**
-     * @dev Returns the description of the attribute.
+     * @dev Remove attribute type `attrId` from `tokenId`.
      */
-    function description(uint256 _attrId) external view returns (string memory);
+    function remove(uint256 tokenId, uint256 attrId) external;
 
     /**
-     * @dev Returns the attributes of the nft.
+     * @dev Increases `amount` value of attribute type `attrId` to `tokenId`.
      */
-    function getNFTAttrs(uint256 _nftId) external view returns (uint256[] memory);
-
-    /**
-     * @dev Returns the max level of the attribute.
-     */
-    function maxLevel(uint256 _attrId) external view returns (uint8);
-
-    /**
-     * @dev Create new attribute.
-     */
-    function create(
-        uint256 _id,
-        string memory _name,
-        string memory _description,
-        uint8 _level,
-        uint8[] memory _probabilities,
-        uint256[] memory _block_intervals
+    function increase(
+        uint256 tokenId,
+        uint256 attrId,
+        uint256 amount
     ) external;
 
     /**
-     * @dev Attach the attribute to NFT.
-     */
-    function attach(uint256 _nftId, uint256 _attrId) external;
-
-    /**
-     * @dev Remove the attribute from NFT.
-     */
-    function remove(uint256 _nftId, uint256 _attrId) external;
-
-    /**
-     * @dev Evolutive the attribute of the NFT.
-     */
-    function evolutive(uint256 _nftId, uint256 _attrId, uint8 _level) external;
-
-    /**
-     * @dev Evolution has the probability to damage the NFT, if it is damaged, it can be repaired.
+     * @dev Decreases `amount` value of attribute type `attrId` from `tokenId`.
      */
-    function repair(uint256 _nftId, uint256 _attrId) external;
-
-    event EvolutiveAttributeCreated(string name, uint256 id);
-    event EvolutiveAttributeAttached(uint256 nftId, uint256 attrId);
-    event EvolutiveAttributeRemoved(uint256 nftId, uint256 attrId);
-    event AttributeEvolutive(uint256 nftId, uint256 attrId, uint8 level);
-    event AttributeRepaired(uint256 nftId, uint256 attrId);
+    function decrease(
+        uint256 tokenId,
+        uint256 attrId,
+        uint256 amount
+    ) external;
 }
-
-
 ```
 
-`evolutive` function
+Updatable attributes of NFTs can change under certain trigger conditions. Such use cases are common in gaming and finance. For example, the life force of an character, combative power of weaponry, debt interest rates, etc.
+
```

Comment:
"the life force of `an` character" - there is a grammar mistake... change an to a